modules/python/init.zsh

```diff
@@ -92,22 +92,31 @@ if (( $+VIRTUALENVWRAPPER_VIRTUALENV || $+commands[virtualenv] )) && \
   # Disable the virtualenv prompt.
   VIRTUAL_ENV_DISABLE_PROMPT=1
 
-  # Enable 'virtualenv' with 'pyenv'.
-  if (( $+commands[pyenv] )) && \
-     pyenv commands | command grep -q virtualenv-init
-  then
-    eval "$(pyenv virtualenv-init -)"
-    # Optionall activate 'virtualenvwrapper' with 'pyenv' is available.
-    if (( $#commands[(i)pyenv-virtualenvwrapper(_lazy|)] )); then
-      pyenv "${${(@O)commands[(I)pyenv-virtualenvwrapper(_lazy|)]}[1]#pyenv-}"
+  if (( $+commands[pyenv] )); then
+    # Grab sorted array of available 'pyenv' plugins by probing 'pyenv'
+    # directly. Scanning shell '$path' isn't enough since they can exist
+    # in 'pyenv' synthesized paths (e.g., '~/.pyenv/plugins') as well.
+    virtenv_plugins=(${(@oM)${(f)"$(pyenv commands --no-sh 2>/dev/null)"}:#virtualenv*})
+
+    # Initialize 'pyenv'
+    if (( $virtenv_plugins[(i)virtualenv-init] <= $#virtenv_plugins )); then
+      eval "$(pyenv virtualenv-init -)"
+    fi
+
+    # Optionally activate 'virtualenvwrapper' plugin when available
+    if (( $virtenv_plugins[(i)virtualenvwrapper(_lazy|)] <= $#virtenv_plugins )); then
+      pyenv "$virtenv_plugins[(R)virtualenvwrapper(_lazy|)]"
     fi
+
+    unset virtenv_plugins
   else
     # Fallback to 'virtualenvwrapper' without 'pyenv' wrapper in '$path'
     # and other known locations on a Debian based system.
     virtenv_sources=(
       ${(@Ov)commands[(I)virtualenvwrapper(_lazy|).sh]}
       /usr/share/virtualenvwrapper/virtualenvwrapper(_lazy|).sh(OnN)
     )
+
     source "${virtenv_sources[1]}"
```

Comment:
FYI, this gets called whether this virtualenvwrapper file exists or not. 

So if you have pyenv installed, but not the virtualenv plugin or the virtualenvwrapper plugin, then you get a"file not found" error at login